app/web/features/search/MapWrapper.tsx

```diff
@@ -0,0 +1,287 @@
+import { addClusteredUsersToMap, layers } from "../search/users";
+import { reRenderUsersOnMap } from "features/search/users";
+import { useCallback, useEffect, useState } from "react";
+import ReplayIcon from "@material-ui/icons/Replay";
+import { Dispatch, SetStateAction } from "react";
+import { UserSearchRes } from "proto/search_pb";
+import TuneIcon from "@material-ui/icons/Tune";
+import { filterData } from "../search/users";
+import { InfiniteData } from "react-query";
+import makeStyles from "utils/makeStyles";
+import { usePrevious } from "utils/hooks";
+import { MutableRefObject } from "react";
+import Button from "components/Button";
+import { User } from "proto/api_pb";
+import Map from "components/Map";
+import { Point } from "geojson";
+import { SEARCH } from "i18n/namespaces";
+import { useTranslation } from "i18n";
+
+const useStyles = makeStyles((theme) => ({
+  searchHereGroup2: {
+    top: "20px",
+    // display: "flex",
+    left: "50%",
+    position: "relative",
+    zIndex: 10,
+  },
+  testContainer: {
+    position: "absolute",
+    left: "50%",
+    width: "100%",
+  },
+  testChildFromGoogle: {
+    width: "300px",
+    top: "30px",
+    display: "flex",
+    position: "relative",
+    left: "-50%",
+    fontSize: " 14px",
+    margin: "8px auto 0",
+    alignItems: "center",
+
+    height: "25px",
+    zIndex: 10,
+  },
+  buttonSearchSettings: {
+    borderRadius: "0 4px 4px 0",
+    "& span": {
+      margin: 0,
+    },
+  },
+  searchHereButton: {
+    borderRadius: "4px 0 0 4px",
+  },
+}));
+
+interface mapWrapperProps {
+  selectedResult:
+    | Pick<User.AsObject, "username" | "userId" | "lng" | "lat">
+    | undefined;
+  isLoading: boolean;
+  locationResult: any;
+  setLocationResult: Dispatch<SetStateAction<any>>;
+  results: InfiniteData<UserSearchRes.AsObject> | undefined;
+  setIsFiltersOpen: Dispatch<SetStateAction<boolean>>;
+  mapInitiallyLocated: boolean;
+  setSelectedResult: Dispatch<
+    SetStateAction<
+      Pick<User.AsObject, "username" | "userId" | "lng" | "lat"> | undefined
+    >
+  >;
+  map: MutableRefObject<MaplibreMap | undefined>;
+}
+
+export default function MapWrapper({
+  map,
+  selectedResult,
+  locationResult,
+  setLocationResult,
+  isLoading,
+  results,
+  mapInitiallyLocated,
+  setSelectedResult,
+  setIsFiltersOpen,
+}: mapWrapperProps) {
+  const { t } = useTranslation([SEARCH]);
+  const [areClustersLoaded, setAreClustersLoaded] = useState(false);
+  const previousResult = usePrevious(selectedResult);
+  const classes = useStyles();
+
+  /**
+   * User clicks on a user on map
+   */
+  const handleMapUserClick = useCallback(
+    (
+      ev: maplibregl.MapMouseEvent & {
+        features?: maplibregl.MapboxGeoJSONFeature[] | undefined;
+      } & EventData
+    ) => {
+      ev.preventDefault();
+
+      const props = ev.features?.[0].properties;
+      const geom = ev.features?.[0].geometry as Point;
```

Comment:
What about instead :

`const geom: Point = ev.features?.[0].geometry;`

My understanding is that will provide more protection, but using `as Point` will just coerce it to `Point` and then you lose the type protection.

Reply:
oh! didn't know about this, will look it up and update it!

Reply:
this is when I start fighting with TS...
![image](https://github.com/user-attachments/assets/7982e28c-f985-48fe-b8c4-c4c9f949a62f)

I get another warning cause `ev.features?.[0].geometry` might be undefined... deleting the `?` also gives me another error, also deleting the `| undefined` from the function param (so doesn't think can be undefined) still gives me another error
![image](https://github.com/user-attachments/assets/5b70bc32-7313-4472-9d9e-c7dfde1f2205)

Reply:
might be worth keeping the code as it was? (also, who created the typing wasn't me, so I'm not sure why the specify that the function param can be undefined